### PR TITLE
gravel: fix inventory gathering, and speed up device probing

### PR DESCRIPTION
**Pull request by jecluis**
Signed-off-by: Joao Eduardo Luis \<joao@suse.com>

### CODE REVIEW

File: src/gravel/controllers/resources/inventory_sub.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+        cb: Callable[[NodeInfoModel], Awaitable[None]],
+        once: bool,
+        registry: List[Subscriber]
+    ):
+        self.cb = cb
+        self.once = once
+        self._registry = registry
+
+    def unsubscribe(self) -> None:
+        self._registry.remove(self)
```
Comment (jhesketh) [inline src/gravel/controllers/resources/inventory_sub.py:35]: Seems odd for Inventory to add the subscriber to the inventory but of the subscriber itself to unsubscribe. I think NodeMgr should call back to the Inventory to remove any subs.

**Review by jhesketh — CHANGES_REQUESTED** [review]
Perhaps a nit.

Also, are we ready to require tests going forward?

**Comment by jecluis** [comment]
@jhesketh addressed, repushed. no tests though, I'll have to figure out how to write those and, tbh, I'd rather have this in soon instead of waiting for them.

**Review by jhesketh — CHANGES_REQUESTED** [review]
The change looks good, but we need to choose here how strict we want to be with requiring tests going forward.

From our discussions it is my impression that we want to require at least some level of testing. This does mean things will move slower and things may be held up more, but I also believe that is the only way we're going to make any progress on growing our test coverage. As such I'm marking this as a "requests changes", but should others disagree the change still looks good and I'm happy for it to merge.

File: src/gravel/tests/unit/controllers/resources/test_inventory.py [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+    mocker: MockerFixture,
+    gstate: GlobalState
+):
+
+    nodeinfo: NodeInfoModel = NodeInfoModel(
+        hostname="foo",
+        model="",
+        vendor="",
+        kernel="123.4",
+        operating_system="fancything",
```
Comment (mgfritch) [inline src/gravel/tests/unit/controllers/resources/test_inventory.py:38]: fancy!

**Review by mgfritch — APPROVED** [review]
lgtm!

**Review by jhesketh — COMMENTED** [review]
Awesome, thanks for adding a test :-)

**Pull request merged**